engine/class_modules/apl/apl_death_knight.cpp

```diff
@@ -95,53 +98,69 @@ void blood( player_t* p )
   precombat->add_action( "fleshcraft" );
 
   default_->add_action( "auto_attack" );
+  default_->add_action( "variable,name=death_strike_dump_amount,value=70,op=setif,condition=covenant.night_fae&buff.deaths_due.remains>6,value_else=55" );
   default_->add_action( "mind_freeze,if=target.debuff.casting.react", "Interrupt" );
-  default_->add_action( "blood_fury,if=cooldown.dancing_rune_weapon.ready&(!cooldown.blooddrinker.ready|!talent.blooddrinker.enabled)" );
-  default_->add_action( "berserking" );
-  default_->add_action( "arcane_pulse,if=active_enemies>=2|rune<1&runic_power.deficit>60" );
-  default_->add_action( "lights_judgment,if=buff.unholy_strength.up" );
-  default_->add_action( "ancestral_call" );
-  default_->add_action( "fireblood" );
-  default_->add_action( "bag_of_tricks" );
-  default_->add_action( "potion,if=buff.dancing_rune_weapon.up", "Since the potion cooldown has changed, we'll sync with DRW" );
+  default_->add_action( "potion,if=buff.dancing_rune_weapon.up","Since the potion cooldown has changed, we'll sync with DRW" );
   default_->add_action( "use_items" );
   default_->add_action( "raise_dead" );
   default_->add_action( "blooddrinker,if=!buff.dancing_rune_weapon.up&(!covenant.night_fae|buff.deaths_due.remains>7)" );
-  default_->add_action( "blood_boil,if=charges>=2&(covenant.kyrian|buff.dancing_rune_weapon.up)" );
-  default_->add_action( "death_strike,if=fight_remains<3" );
+  default_->add_action( "marrowrend,if=covenant.necrolord&buff.bone_shield.stack<=0","Marrowrend on pull if Necrolord in order to guarantee ossuary during the first DRW with AL" );
+  default_->add_action( "call_action_list,name=racials" );
+  default_->add_action( "sacrificial_pact,if=(!covenant.night_fae|buff.deaths_due.remains>6)&!buff.dancing_rune_weapon.up&(pet.ghoul.remains<2|target.time_to_die<gcd)", "Attempt to sacrifice the ghoul if we predictably will not do much in the near future" );
   default_->add_action( "call_action_list,name=covenants" );
+  default_->add_action( "blood_tap,if=(rune<=2&rune.time_to_4>gcd&charges_fractional>=1.8)|rune.time_to_3>gcd" );
+  default_->add_action( "dancing_rune_weapon,if=!covenant.venthyr" );
+  default_->add_action( "run_action_list,name=drw_up,if=buff.dancing_rune_weapon.up&!covenant.venthyr" );
+  default_->add_action( "dancing_rune_weapon,if=covenant.venthyr&cooldown.swarming_mist.ready&runic_power>=(90-(spell_targets.swarming_mist*3))" );
```

Comment:
might be able to clean things up further by breaking these out into their own cooldowns action list which can be called before covenants, letting you keep stuff like each marrowrend line next to each other.

Reply:
Sounds good! I'll look into it.